### PR TITLE
[dashboard] Use public-api to CreateTeams

### PR DESCRIPTION
**Pull request by easyCZ**
## Related Issue(s)

* Depends on https://github.com/gitpod-io/gitpod/pull/14152
* Fixes https://github.com/gitpod-io/gitpod/issues/14309

The feature is enabled behind a feature flag.

## How to test

* Use preview
* Create a new team (with DevTools open)
* Observe the request goes to `api.<domain>`
* Validate create workflow continues to behave the same

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

**Comment by easyCZ** [comment]
/werft run with-clean-slate-deployment=true with-preview=true

**Comment by easyCZ** [comment]
/werft run with-clean-slate-deployment=true with-preview=true

File: components/dashboard/src/projects/NewProject.tsx [hunk 1]
```diff
@@ -729,6 +732,37 @@ function NewTeam(props: { onSuccess: (team: Team) => void }) {
...
             return;
         }
+
+        const isExperimentalTeamsServiceEnabled = await getExperimentsClient().getValueAsync(
+            "publicApiExperimentalTeamsService",
+            false,
+            {
+                user,
+            },
+        );
```
Comment (andrew-farries) [inline components/dashboard/src/projects/NewProject.tsx:742]: I think we can use the `FeatureFlagContext` to get the value of this feature flag here.

File: components/dashboard/src/service/public-api.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            scopes: [
+                "function:getGitpodTokenScopes",
+                "function:getWorkspace",
+                "function:getWorkspaces",
+                "function:createTeam",
+                "function:joinTeam",
+                "function:getTeamMembers",
+                "function:getGenericInvite",
+                "function:listenForWorkspaceInstanceUpdates",
+                "resource:default",
```
Comment (andrew-farries) [inline components/dashboard/src/service/public-api.ts:28]: Why is it necessary to generate a new token here? Won't the user always have a suitable token when performing this action?

File: components/dashboard/src/teams/NewTeam.tsx [hunk 3]
```diff
@@ -8,15 +8,41 @@ import { FormEvent, useContext, useEffect, useState } from "react";
...
     let name = "";
     const createTeam = async (event: FormEvent) => {
         event.preventDefault();
+        const isExperimentalTeamsServiceEnabled = await getExperimentsClient().getValueAsync(
+            "publicApiExperimentalTeamsService",
+            false,
+            {
+                user,
+            },
+        );
```
Comment (andrew-farries) [inline components/dashboard/src/teams/NewTeam.tsx:30]: Same comment about using the `FeatureFlagContext`.

**Review by andrew-farries — CHANGES_REQUESTED** [review]

File: components/dashboard/src/service/public-api.ts (hunk 2 above)
Comment (easyCZ) [reply components/dashboard/src/service/public-api.ts:28]: They have a cookie based session which is not compatible with the AccessToken based auth on Public API. I agree this is not great, or pretty, but it's a temporary solution until we have first class Access Tokens.

File: components/dashboard/src/projects/NewProject.tsx (hunk 1 above)
Comment (easyCZ) [reply components/dashboard/src/projects/NewProject.tsx]: Good shout. Updated.

**Review by andrew-farries — APPROVED** [review]

**Comment by easyCZ** [comment]
Thanks for the swift review @andrew-farries!

**Pull request merged**